frontend/src/app/max-validator.directive.ts

```diff
@@ -0,0 +1,50 @@
+/* tslint:disable:directive-selector */
+import { Directive, forwardRef, HostBinding, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { AbstractControl, NG_VALIDATORS, ValidationErrors, Validator, ValidatorFn, Validators } from '@angular/forms';
+
+/**
+ * Provider which adds MaxValidatorDirective to NG_VALIDATORS.
+ */
+export const MAX_VALIDATOR: any = {
+  provide: NG_VALIDATORS,
+  useExisting: forwardRef(() => MaxValidatorDirective),
+  multi: true
+};
+
+/**
+ * A directive which installs the MaxValidator for any `formControlName`,
+ * `formControl`, or control with `ngModel` that also has a `max` attribute.
+ */
+@Directive({
+  selector: '[max][type=number][formControlName],[max][type=number][formControl],[max][type=number][ngModel]',
+  providers: [MAX_VALIDATOR]
+})
+export class MaxValidatorDirective implements Validator, OnChanges {
+  private _validator: ValidatorFn;
+  private _onChange: () => void;
+
+  @Input() max: number;
+
+  @HostBinding('attr.max') get maxAttr() {
+    return this.max !== null && this.max !== undefined ? `${this.max}` : null;
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if ('max' in changes) {
```

Comment:
nitpicking: A setter would be enough?

Reply:
Yes, will change. I copied and pasted from angular directives.